src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -791,15 +791,19 @@ private function getLinkObject(string $resourceClass, string $operationId, strin
                 continue;
             }
 
+            $propertyName = $propertyMetadata->getSerializedName() ?? $propertyName;
             $linkObject['parameters'][$propertyName] = sprintf('$response.body#/%s', $propertyName);
             $identifiers[] = $propertyName;
         }
 
         if (!$linkObject) {
             return [];
         }
+        preg_match_all('/{(.*)}/', $path, $uriParams);
+        $identifierUriParam = !empty($uriParams[1]) ? end($uriParams[1]) : 'id';
```

Comment:
I need to take a look at this as I think it's related to my work on identifiers. We'll definitely merge this at some point!

Reply:
Hi @soyuka! Don't hesitate to ask me if you need some clarification or if you think any change is needed to avoid conflicts with your work on identifiers.

Reply:
Isn't there another way to get identifiers here ? I'm not fond of the regex to fetch params. Also, an url param is not always an identifier. 
I like the idea but maybe it'll be easier to add on master as a new feature once I can work on my backward compat on #3664

Reply:
Hi @soyuka. 

You are right. This was a workaround so it should be fine on most cases (at least in all the cases the path is generated by API Platform, even if the identifier property has a custom serialized name), but you are totally right it is not a permanent fix and would fail in some edge cases with custom paths on non restful apis.

As we got a path string as parameter in getLinkObject function, I don't think we can find a real solution to this problem in 2.5, as any identifier url parameter should be done on a string guessing.

I think the best solution in this scenario, and until #3664 is merged and this issue can be revisited, is to omit the parameter name in the Links strings, changing it to ```The `%1$s` value returned in the response can be used as the Resource identifier parameter in `GET %3$s`.```

With that solution, we can merge this bugfix in 2.5 and search for a way to get the identifier parameter name in paths in master as a new feature.

Reply:
You can extract the identifiers by looping on `PropertyCollection` and finding using `isIdentifier` or using the IdentifiersExtractor class.

Reply:
Getting the identifiers is only half the problem. In you use a custom path or set a route_name in the operation, the identifier parameter name in the URL could be different than the identifier property name on the Resource.

In getLinkObject we actually have an array of the identifiers, so I though to check if the $identifierUriParam extracted from the URI was named as the $identifiers[0] used in Links text. But it did't work, as the parameter don't have to be called as the identifier.

I'm not sure if we could guarantee that any parameter in the URL match to the resource identifier, as in Symfony routes don't exist any "real" mapping between parameters and entities/Resource properties, they are only a placeholder as @dunglas said.

Without any breaking change, I think the only way to not to fail in any case (even with a custom serialized name for the resource identifier or a custom path/route with an  URL parameter called different than the identifier property) is to omit the URL parameter name in the Links text.

Reply:
So let's just use `id` hard-coded. I'll fix this later properly when we'll have a way to specify identifiers.

I'm really not fond of the preg_match and if someone needs to change it there's always the possibility of overriding using a custom swagger normalizer. It'll introduce less complexity. Note that the code needs to be adapted to master where OpenAPI is not handled the same way but I'd like to use this serialized_name.

Reply:
@soyuka so should I open another pull request adapting this to master? Or should we wait until you fix the identifiers problem?

In this pull request I will revert to the hardcoded `id` and will squash the commits in only one commit rebased to current 2.5.

Reply:
>  should I open another pull request adapting this to master? 

Sure! 

> In this pull request I will revert to the hardcoded id and will squash the commits in only one commit rebased to current 2.5.

You don't really need to squash we can do this when merging :), as you want